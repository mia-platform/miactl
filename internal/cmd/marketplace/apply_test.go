// Copyright Mia srl
// SPDX-License-Identifier: Apache-2.0
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package marketplace

import (
	"net/http"
	"net/http/httptest"
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func applyMockServer(t *testing.T, statusCode int) *httptest.Server {
	t.Helper()
	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		var isReqOk = assert.Equal(t, applyEndpoint, r.RequestURI) && assert.Equal(t, http.MethodPost, r.Method)
		if !isReqOk {
			w.WriteHeader(http.StatusNotFound)
			require.Fail(t, "unsupported call")
			return
		}
		w.WriteHeader(statusCode)
	}))
}

func TestApplyResourceCmd(t *testing.T) {
	t.Run("test command creation", func(t *testing.T) {
		t.Skip()
	})
}
