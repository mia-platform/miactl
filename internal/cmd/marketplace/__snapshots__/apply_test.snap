
[TestApplyPrintApplyOutcome/should_match_snapshot_with_both_valid_files_and_validation_errors - 1]
3 of 4 items have been successfully applied:

  ITEM ID  NAME         STATUS    

  id1      some name 1  Updated   
  id2      some name 2  Inserted  
  id3      some name 3  Inserted  

1 of 4 items have not been applied due to validation errors:

  ITEM ID  NAME         VALIDATION ERRORS      

  id4      some name 4  some validation error  

---

[TestApplyPrintApplyOutcome/should_match_snapshot_with_validation_errors_only - 1]
2 of 2 items have not been applied due to validation errors:

  ITEM ID  NAME         VALIDATION ERRORS               

  id3      some name 3  some validation error           
  id4      some name 4  some other validation error     
                        and also some other validation  
                        error                           

---

[TestApplyPrintApplyOutcome/should_match_snapshot_with_valid_files_only - 1]
3 of 3 items have been successfully applied:

  ITEM ID  NAME         STATUS    

  id1      some name 1  Updated   
  id2      some name 2  Inserted  
  id3      some name 3  Inserted  

---
